tutorials/vue/vue-d3.md

```diff
@@ -0,0 +1,148 @@
+[D3.js](https://d3js.org/) is a great library for visualizing data and displaying it in your projects. This library gives you the building blocks to make very aesthetically pleasing charts in projects. To use it
+with your Vue 3 projects, you can download a libray called `vue-d3-charts`
+which is built upon `D3.js`. `vue-d3-charts` does not support vue2 and so
+you will need to import `D3.js`.
+
+As of this writing, development for vue-d3-charts is currently frozen as the head maintainer does not have the time
+to continue improving this library. However, individuals are free to contribute [here](https://github.com/Saigesp/vue-d3-charts).
+We recommend using other data visualizing libraries as any issues you may run into while using
+this library will most likely not be solved for a good period of time.
+
+Here is an example of using `D3.js` in Vue2 to create a bar graph.
+A great thing about `D3.js` is that you can import and parse a csv file from an online source to use directly in your visualizations.
+Below is the code:
+
+```html
+<script src="https://d3js.org/d3.v6.js"></script>
+<script src="https://unpkg.com/vue"></script>
+<div class = "app">{{message}}
+<bargraph></bargraph>
+</div>
+<script type = "text/javascript">
+Vue.component('bargraph', {
+    mounted(){
+        this.retrieveGraph();
+    },
+    methods: {
+        retrieveGraph: function(){
+            let margin = ({top: 30, right: 30, bottom: 70, left: 60});
+            let width = 460 - margin.left - margin.right;
+            let height = 400 - margin.top - margin.bottom;
+            var svg = d3.select("#example")
```

Comment:
Is there any way to use [vue refs](https://masteringjs.io/tutorials/vue/refs) to handle this instead? Selecting DOM elements by id in component methods is generally not best practice.